File: server/Models/Report.cs
namespace help_reviews.Models;

public class Report : RepoItem<int>
{
  public string Title { get; set; }
  public string Body { get; set; }
  public string PictureOfDisgust { get; set; }
  public string CreatorId { get; set; }
  public int RestaurantId { get; set; }
  public Profile Creator { get; set; }
}

File: client/src/models/Report.js
import { DatabaseItem } from "./DatabaseItem.js";
import { Profile } from "./Profile.js";


export class Report extends DatabaseItem {
  constructor(data) {
    super(data);
    this.title = data.title;
    this.body = data.body;
    this.pictureOfDisgust = data.pictureOfDisgust;
    this.creatorId = data.creatorId;
    this.restaurantId = data.restaurantId;
    this.creator = new Profile(data.creator);
  }
}
